pkg/kubelet/images/image_gc_manager.go

```diff
@@ -237,8 +252,14 @@ func (im *realImageGCManager) detectImages(ctx context.Context, detectTime time.
 	// Make a set of images in use by containers.
 	for _, pod := range pods {
 		for _, container := range pod.Containers {
-			klog.V(5).InfoS("Container uses image", "pod", klog.KRef(pod.Namespace, pod.Name), "containerName", container.Name, "containerImage", container.Image, "imageID", container.ImageID)
-			imagesInUse.Insert(container.ImageID)
+			if !isRuntimeClassInImageCriAPIEnabled {
+				klog.V(5).InfoS("Container uses image", "pod", klog.KRef(pod.Namespace, pod.Name), "containerName", container.Name, "containerImage", container.Image, "imageID", container.ImageID)
+				imagesInUse.Insert(container.ImageID)
+			} else {
+				imageKey := getImageTuple(container.ImageID, container.ImageRuntimeHandler)
+				klog.V(5).InfoS("Container uses image", "pod", klog.KRef(pod.Namespace, pod.Name), "containerName", container.Name, "containerImage", container.Image, "imageID", container.ImageID, "imageKey", imageKey)
+				imagesInUse.Insert(imageKey)
+			}
```

Comment:
same logging comment here: 

```suggestion
                        imageKey := container.ImageID
			if isRuntimeClassInImageCriAPIEnabled {
				imageKey := getImageTuple(container.ImageID, container.ImageRuntimeHandler)
			}
			
			klog.V(5).InfoS("Container uses image", "pod", klog.KRef(pod.Namespace, pod.Name), "containerName", container.Name, "containerImage", container.Image, "imageID", container.ImageID, "imageKey", imageKey)
			imagesInUse.Insert(imageKey)
```

Reply:
Wanted to explicitly call out the difference between imageID and imageKey. That is why I didn't change these two if-else blocks:
![image](https://github.com/kubernetes/kubernetes/assets/99994218/fc6fcbf8-98f9-492a-bdce-757e3c11daf4)